apache-rat-core/src/main/java/org/apache/rat/config/parameters/Description.java

```diff
@@ -0,0 +1,321 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.rat.config.parameters;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.stream.Collectors;
+
+import org.apache.rat.ConfigurationException;
+import org.apache.rat.analysis.IHeaderMatcher;
+import org.apache.rat.config.parameters.Component.Type;
+import org.apache.rat.utils.Log;
+
+/**
+ * A description of a component.
+ */
+public class Description {
+    /** The type of component this describes */
+    private final Type type;
+    /**
+     * The common name for the component. Set by ConfigComponent.name() or
+     * class/field name.
+     */
+    private final String name;
+    /** The description for the component */
+    private final String desc;
+    /** The class of the getter/setter parameter */
+    private final Class<?> childClass;
+    /** True if the getter/setter expects a collection of childClass objects */
+    private final boolean isCollection;
```

Comment:
When I read this the first time I instantly thought of isRecursive(), but it would imply a great many changes.